rust-runtime/aws-smithy-observability/src/provider.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Definitions of high level Telemetry Providers.
+
+use std::sync::Arc;
+
+use crate::{meter::MeterProvider, noop::NoopMeterProvider};
+
+/// A struct to hold the various types of telemetry providers.
+#[non_exhaustive]
+pub struct TelemetryProvider {
```

Comment:
Any reason to not make it a trait? What was your rationale for or against?

Reply:
Mostly because it is not specified in the SRA and so isn't really an interface we would expect users to be implementing themselves. Since the `TelemetryProvider` just exists to store and return references to the `MeterProvider` (and eventually maybe a `TracerProvider` and `LogProvider`) it doesn't seem like something a user would need to implement any custom functionality on.